### PR TITLE
Use stdint.h for integer types in arrays

### PR DESCRIPTION
**Pull request by EmilyBourne**
As discussed in PR #548, we are now using stdint.h for integer types in C. This PR makes those changes for arrays in C.

cc: @saniQQ

### CODE REVIEW

**Review by saniQQ — COMMENTED** [review]
is it necessary to use precise types for index variables?

File: tests/ndarrays/test_ndarrays.c [hunk 1]
```diff
@@ -1,459 +1,459 @@
...
+}
+
+int32_t test_indexing_int32(void)
+{
+    int32_t m_1[] = {2, 3, 5, 5, 6, 7, 10, 11,
+                12, 260, 6, 8, 8, 0, 45, 0,
+                1, 0, 0, 1, 200, 33, 5, 57,
+                62, 70, 103, 141, 122, 26, 36, 82,
+                8, 10, 4115, 22, 1, 11, 1, 19};
+    int32_t m_1_shape[] = {5, 8};
```
Comment (rakati) [inline tests/ndarrays/test_ndarrays.c:88]: why you use a precise type shape?
I think you need t use only int, as mentioned in the header file.

File: tests/ndarrays/test_ndarrays.c [hunk 2]
```diff
@@ -1,459 +1,459 @@
...
+    t_ndarray x;
+    int32_t index;
+    int32_t c_index;
```
Comment (rakati) [inline tests/ndarrays/test_ndarrays.c:91]: same here

File: tests/ndarrays/test_ndarrays.c [hunk 3]
```diff
@@ -1,459 +1,459 @@
...
+}
+
+int32_t test_indexing_double(void)
+{
+    double m_1[] = {2, 3, 5, 5, 6, 7, 10, 11,
+                    12, 260, 6.34, 8, 8.002, 0.056, 45, 0.1,
+                    1.02, 0.25, 0.00005, 1, 200, 33, 5, 57,
+                    62, 70, 103.009, 141, 122, 26.50, 36.334, 82,
+                    8.44002, 10.056, 4115, 22.1, 1.1102, 011.25, 1.01110005, 19};
+    int32_t m_1_shape[] = {5, 8};
```
Comment (rakati) [inline tests/ndarrays/test_ndarrays.c:117]: same here

File: tests/ndarrays/test_ndarrays.c [hunk 4]
```diff
@@ -1,459 +1,459 @@
...
+}
+
+int32_t test_indexing_cdouble(void)
+{
+    double complex m_1[] = {0.37 + 0.588*I,  0.92689451+0.57106791*I,
+                            0.93598206+0.30289964*I,  0.54404246+0.09516331*I,
+                            0.02827254+0.00432899*I,  0.06873651+0.24810741*I,
+                            0.94040543+0.43508215*I,  0.58532094+0.67890618*I,
+                            0.68742283+0.64951155*I,  0.15372315+0.89699101*I};
+    int32_t m_1_shape[] = {5, 2};
```
Comment (rakati) [inline tests/ndarrays/test_ndarrays.c:146]: same here

**Review by rakati — COMMENTED** [review]
> is it necessary to use precise types for index variables?

no, just simple int.
in the forloops also and shapes.

**Review by saniQQ — APPROVED** [review]

**Comment by EmilyBourne** [comment]
@saniQQ @rakati All your questions are about precision so I will answer them together here. As mentioned in the PR description, after PR #548 we will be using precise types in all the code generated by pyccel. While technically shapes don't need to have precise types, if they don't there may be casts generated when generating the code that creates the shape. The same is also true for indexes. (I don't know on what, if any, system this might be the case as all systems I know of have int==int32_t, but there is no guarantee of this so it is better to be consistent)

File: tests/ndarrays/test_ndarrays.c (hunk 1 above)
Comment (EmilyBourne) [reply tests/ndarrays/test_ndarrays.c:181]: You are right, it seems the header still needs some fixing

**Review by rakati — APPROVED** [review]

**Review by yguclu — APPROVED** [review]

**Pull request merged**